### PR TITLE
added undev .cfg for Hyperkin Trooper V2

### PR DESCRIPTION
**Pull request by Rstation4**
"Hyperkin "Trooper II" Premium Atari 2600 Style Wired USB Joystick Game Controller"

### CODE REVIEW

**Comment by hizzlekizzle** [comment]
While we have you, would you mind adding button labels, as well?

**Comment by Rstation4** [comment]
> While we have you, would you mind adding button labels, as well?

As in the the file with a comment # Explaining which button is which. The game pad has 6 buttons. a Select, Star, Blue Button 1, Blue Button 2, A LB button and a RB button along with the D-pad joystick.

gamepad

**Comment by Rstation4** [comment]
How about this for example

input_driver = "udev"
input_device = "Hyperkin Trooper V2"
input_device_display_name = "Hyperkin Trooper V2"
input_vendor_id = "11812"
input_product_id = "14442"

# Left blue button number 1
input_b_btn = "0"
input_select_btn = "4"
input_start_btn = "5"
input_up_axis = "-1"
input_down_axis = "+1"
input_left_axis = "-0"
input_right_axis = "+0"

# Right blue button number 2
input_a_btn = "1"
# Left bumper/shoulder button
input_l_btn = "2"
# Right bumper/shoulder button
input_r_btn = "3"

**Comment by hizzlekizzle** [comment]
You can see an example of button labels here: https://github.com/libretro/retroarch-joypad-autoconfig/blob/master/udev/Sony-PlayStation3-DualShock3-Controller-Bluetooth.cfg#L31

It's just a few lines that will make the retropad mappings show the name from the actual controller in the quick menu > controls menu.

**Comment by Rstation4** [comment]
> You can see an example of button labels here: https://github.com/libretro/retroarch-joypad-autoconfig/blob/master/udev/Sony-PlayStation3-DualShock3-Controller-Bluetooth.cfg#L31
> 
> It's just a few lines that will make the retropad mappings show the name from the actual controller in the quick menu > controls menu.

The blue buttons Should I label it "Blue Button 1" and the other "Blue Button 2" its one of those joypads that dont use a number, a Alphabet or a symbol of anykind on the gamepad.

**Comment by hizzlekizzle** [comment]
Personally, I would probably name them something like "left button" and "right button" to go with the "left bumper" / "right bumper" nomenclature, unless there's any engravings/stickers that actually denote which button is which. Really, it's pretty loosey-goosey; just name them something that will make sense to someone holding the controller in their hands :)

File: udev/Hyperkin_Trooper_V2.cfg [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+input_up_axis = "-1"
+input_down_axis = "+1"
+input_left_axis = "-0"
+input_right_axis = "+0"
+input_a_btn = "1"
+input_l_btn = "2"
+input_r_btn = "3"
+
+input_b_btn_label = "Blue Button (Left)"
+input_a_btn_label = "Blue Button (Right)"
```
Comment (RobLoach) [inline udev/Hyperkin_Trooper_V2.cfg:18]:
``` suggestion
input_b_btn_label = "Left Button"
input_a_btn_label = "Right Button"
```

File: udev/Hyperkin_Trooper_V2.cfg [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+input_select_btn_label = "Select"
+input_start_btn_label = "Start"
+input_up_axis_label = "Up"
+input_down_axis_label = "Down"
+input_left_axis_lbel = "Left"
+input_right_axis_label = "Right"
+input_l_btn_label = "Left Shoulder/Bumper"
+input_r_btn_label = "Right Shoulder/Bumper"
```
Comment (RobLoach) [inline udev/Hyperkin_Trooper_V2.cfg:26]: input_l_btn_label = "Left Shoulder"
input_r_btn_label = "Right Shoulder"

**Comment by RobLoach** [comment]
> go with the "left bumper" / "right bumper" nomenclature

While I don't see any engravings in pictures, I do see those two buttons referred to as "shoulder buttons" in the specs on some of the sites out there: https://www.microcenter.com/product/627549/hyperkin-%e2%80%9ctrooper-ii-premium-atari-2600-style-wired-usb-controller-for-pc-mac-raspberry-pi-retro-pie-linux-joystick

File: udev/Hyperkin_Trooper_V2.cfg [hunk 3]
```diff
@@ -0,0 +1,26 @@
+input_driver = "udev"
```
Comment (RobLoach) [inline udev/Hyperkin_Trooper_V2.cfg:1]:
``` suggestion
# Hyperkin Trooper 2 Premium Controller
# https://www.hyperkin.com/supaboy-sfc-portable-pocket-console-for-snes-super-famicom-hyperkin.html
input_driver = "udev"
```

**Pull request merged**